pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -394,3 +399,109 @@ func (ss *InferenceServiceStatus) ClearCondition(conditionType apis.ConditionTyp
 		conditionSet.Manage(ss).ClearCondition(conditionType)
 	}
 }
+
+func (ss *InferenceServiceStatus) UpdateModelRevisionStates(modelState ModelState, info *FailureInfo) {
+	if ss.ModelStatus.ModelRevisionStates == nil {
+		ss.ModelStatus.ModelRevisionStates = &ModelRevisionStates{ActiveModelState: modelState}
+	} else {
+		ss.ModelStatus.ModelRevisionStates.ActiveModelState = modelState
```

Comment:
here should be `TargetModelState` as we are passing the status from `latestCreatedRevision`. `ActiveModelState` is equivalent to `latestReadyRevision` and most of time should be `loaded` when it is not empty, when `latestReadyRevison` is empty and there is only one revision, only `TargetModelState` should be populated. Things get a bit tricky for the raw deployment case, as it is doing rolling upgrade when model is updated, it is hard to differentiate between `ActiveModel` and `TargetModel`.